ui/components/app/import-token/network-filter-import-token/network-filter-dropdown/index.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+  CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP,
+  FEATURED_NETWORK_CHAIN_IDS,
+} from '../../../../../../shared/constants/network';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+
+type NetworkFilterDropdownProps = {
+  title: string;
+  buttonDataTestId: string;
+  isCurrentNetwork: boolean;
+  openListNetwork: () => void;
+  currentNetworkImageUrl: string;
+  allOpts: Record<string, boolean>;
+  isDropdownOpen: boolean;
+  setIsDropdownOpen: (isOpen: boolean) => void;
+  dropdownRef: React.RefObject<HTMLDivElement>;
+};
+
+export const NetworkFilterDropdown = ({
+  title,
+  buttonDataTestId,
+  isCurrentNetwork,
+  openListNetwork,
+  currentNetworkImageUrl,
+  allOpts,
+  isDropdownOpen,
+  setIsDropdownOpen,
+  dropdownRef,
+}: NetworkFilterDropdownProps) => {
+  const t = useI18nContext();
+
+  const renderItem = () => {
+    if (isCurrentNetwork) {
+      return (
+        <Box
+          display={Display.Flex}
+          justifyContent={JustifyContent.spaceBetween}
+          width={BlockSize.Full}
+          paddingTop={3}
+          paddingBottom={3}
+          gap={3}
+        >
+          <Box>
+            <Text
+              variant={TextVariant.bodyMdMedium}
+              color={TextColor.textDefault}
+            >
+              {t('currentNetwork')}
+            </Text>
+          </Box>
+          <Box
+            display={Display.Flex}
+            alignItems={AlignItems.flexStart}
+            onClick={openListNetwork}
+          >
+            <AvatarNetwork
+              key={currentNetworkImageUrl}
+              name={currentNetworkImageUrl ?? ''}
+              src={currentNetworkImageUrl ?? undefined}
+              size={AvatarNetworkSize.Sm}
+            />
+          </Box>
+        </Box>
+      );
+    }
+    return (
+      <Box
+        display={Display.Flex}
+        justifyContent={JustifyContent.spaceBetween}
+        width={BlockSize.Full}
+        paddingTop={3}
+        paddingBottom={3}
+        gap={3}
+      >
+        <Text variant={TextVariant.bodyMdMedium} color={TextColor.textDefault}>
+          {t('popularNetworks')}
+        </Text>
+        <Box
+          display={Display.Flex}
+          alignItems={AlignItems.flexEnd}
+          onClick={openListNetwork}
+        >
+          {FEATURED_NETWORK_CHAIN_IDS.filter((chain) => allOpts[chain]).map(
+            (chain, index) => {
+              const networkImageUrl = CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP[chain];
+              return (
+                <AvatarNetwork
+                  key={networkImageUrl}
+                  name={networkImageUrl}
+                  src={networkImageUrl ?? undefined}
+                  size={AvatarNetworkSize.Sm}
+                  style={{
+                    marginLeft: index === 0 ? 0 : '-20px',
+                    zIndex: 5 - index,
+                  }}
+                />
+              );
+            },
+          )}
+        </Box>
+      </Box>
+    );
+  };
```

Comment:
Hmm, I'm still a bit worried about this `renderItem` function.

If you don't think it should be a separate memoized component, should we at least put this into a `useMemo` hook? It looks like it depends on specific data, and I think it would be helpful know what data this function/component rely on to only need to re-render when this data changes.